fc/parsing/actions.py

```diff
@@ -1052,111 +1220,34 @@ def modify_model(self, model, units):
             # TODO: Look for variable annotated as time instead?
             raise ProtocolError('Model must contain at least one ODE.')
 
-        # Convert free variable units
-        self._convert_time_if_needed()
-
-        # Ensure inputs exist, adding them if needed, and storing initial values set by user.
-        # This also performs unit conversion where needed for existing input variables.
-        self._add_input_variables()
+        # Annotate all state variables with the magic `oxmeta:state_variable` term. This is done before unit conversion
+        # so that annotations are transferred where needed. The original order in which state variables were defined is
+        # stored.
+        original_state_order = self._annotate_state_variables()
 
-        # Add variables created by ``var`` statements
-        self._add_local_variables()
+        # Collate information about variables mentioned in protocol.
+        self._collate_variable_information()
 
-        # Add variables created by ``define`` statements - must also be listed as output or optional input
-        self._add_define_variables()
+        # Convert units of model variables, where needed.
+        self._convert_time_unit_if_needed()
+        self._convert_units()
 
-        # TODO: Add variables created by ``optional`` statements with a ``default`` clause
-
-        # Now that all variables are in place, it's possible to create the sympy equations
-        self._convert_equations_to_sympy()
-
-        # With variables and equations in place, we can perform sanity checks on model-protocol combination before
-        # making any modifications
-        self._sanity_check()
-
-        # Process ``define`` statements, adding or replacing equations where needed,
-        # and setting any initial values defined through inputs.
-        self._add_or_replace_equations()
+        # Processes the variables used by the protocol, creating variables if needed and updating model equations.
+        self._process_protocol_variables()
 
         # Process ``clamp`` statements with an RHS
         self._handle_clamping()
 
-        self._annotate_state_variables()
+        # Process vector outputs
+        self._process_transitive_variables()
 
-        # Convert units for output variables
-        output_variables = self._convert_output_units()
+        # Gather state variables, store vector ordering and add to ProtocolVariable if requested
```

Comment:
```suggestion
        # Gather state variables, store vector ordering and add to protocol variables if requested
```

Reply:
Nope! The magic ProtocolVariable is already in the list of protocol variables. This is about adding the list of state variables to the magic ProtocolVariable

Reply:
Ah, I see!